apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx

```diff
@@ -71,7 +71,11 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
   let booking: GetBookingType | null = null;
   if (rescheduleUid) {
     booking = await getBookingForReschedule(`${rescheduleUid}`, session?.user?.id);
-    if (booking?.status === BookingStatus.CANCELLED && !allowRescheduleForCancelledBooking) {
+    if (
+      booking?.status === BookingStatus.CANCELLED &&
+      !allowRescheduleForCancelledBooking &&
+      !eventData.allowReschedulingCancelledBookings
+    ) {
```

Comment:
This logic is buggy but it is the behaviour in production as well. Bascailly we are allowing the reschedule in UI for cancelled booking but the actual reschedule is still rejected by backend.